sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java

```diff
@@ -82,6 +84,7 @@ private Exporter(InternalState internalState) {
     @SuppressWarnings("BooleanParameter")
     public void run() {
```

Comment:
It looks like it'd be better to refactor to introduce `CompletableResultCode doRun` and call `join` when shutting down. I suspect much would change when introducing `forceFlush` to metrics but in the meantime that seems like a reasonable workaround.

Reply:
I agree, I think we should use CompletableResultCode here to match the behavior on the tracing side.

Reply:
@anuraaga your suggestion was to create a `doRun` method that returned `CompleteableResultCode` so that the `shutdown` implementation can join that result code, right? Otherwise, this `run` method is async if the exporter is async and a proper orderly shutdown couldn't be implemented.

Reply:
I see what you mean here. Thanks for the explanation. :)

Reply:
Can't this `run()` method just call `doRun()` and ignore the result, rather than duplicating all the code?

Reply:
or maybe not ignore the result, but `join` it, so that this remains synchronous, I guess.

Reply:
I will let this `run()` directly call `doRun()` and ignore the result so that `run()` could keep asynchronous. I will let `shutdown()` call `doRun()` and `join()` the result to make sure it's an orderly shutdown.